tree/bst.go

```diff
@@ -160,3 +160,117 @@ func (n *node) getBSTNodeWithValue(val int) *node {
 
 	return n.right.getBSTNodeWithValue(val)
 }
+
+// changes By Prerna 08/01/2022
+
+// 1. Is validate BST
+func (t *BinarySearchTree) ValidBST() {
```

Comment:
As discussed, this should be implemented on Binary tree.

Reply:
Also the return should be bool. So that it can be used outside the package to make decisions.